rmw_implementation/src/functions.cpp

```diff
@@ -61,17 +64,18 @@ get_library()
         ("failed to find shared library of rmw implementation. Searched " + env_var).c_str());
       return nullptr;
     }
+
     try {
-      lib = new Poco::SharedLibrary(library_path);
-    } catch (Poco::LibraryLoadException & e) {
+      lib = std::make_shared<rcpputils::SharedLibrary>(library_path.c_str());
+    } catch (const std::runtime_error & e) {
       RMW_SET_ERROR_MSG(
-        ("failed to load shared library of rmw implementation. Exception: " +
-        e.displayText()).c_str());
+        ("failed to allocate memory " + library_path + ": " + std::string(
+          e.what())).c_str());
       return nullptr;
-    } catch (...) {
-      RMW_SET_ERROR_MSG("failed to load shared library of rmw implementation");
+    } catch (const std::bad_alloc & e) {
       RMW_SET_ERROR_MSG(
-        ("failed to load shared library of rmw implementation: " + library_path).c_str());
+        ("Cannot open library " + library_path + ": " + std::string(
```

Comment:
Why change the static part of the error message? The new string doesn't mention "rmw implementation" at all. If there is no reason to change please keep the static part of the message as it was before.

Reply:
Restored